### PR TITLE
Filter classes loaded by scripts

### PR DESCRIPTION
**Pull request by rmuir**
Since 2.2 we run all scripts with minimal privileges, similar to applets in your browser.
The problem is, they have unrestricted access to other things they can muck with (ES, JDK, whatever).
So they can still easily do tons of bad things.

This PR restricts what classes scripts can load via the classloader mechanism, to make life more difficult.
The "standard" list was populated from the old list used for the groovy sandbox: though
a few more were needed for tests to pass (java.lang.String, java.util.Iterator, nothing scary there).

Additionally, each scripting engine typically needs permissions to some runtime stuff.
That is the downside of this "good old classloader" approach, but I like the transparency and simplicity,
and I don't want to waste my time with any feature provided by the engine itself for this, I don't trust them.

This is not perfect and the engines are not perfect but you gotta start somewhere. For expert users that
need to tweak the permissions, we already support that via the standard java security configuration files, the specification is simple, supports wildcards, etc (though we do not use them ourselves).

### CODE REVIEW

File: plugins/lang-python/src/test/java/org/elasticsearch/script/python/PythonSecurityTests.java [hunk 1]
```diff
@@ -66,12 +68,12 @@ private void assertFailure(String script) {
             doTest(script);
             fail("did not get expected exception");
         } catch (PyException expected) {
-            Throwable cause = expected.getCause();
             // TODO: fix jython localization bugs: https://github.com/elastic/elasticsearch/issues/13967
-            // this is the correct assert:
-            // assertNotNull("null cause for exception: " + expected, cause);
-            assertNotNull("null cause for exception", cause);
-            assertTrue("unexpected exception: " + cause, cause instanceof SecurityException);
+            // we do a gross hack for now
```
Comment (rjernst) [inline plugins/lang-python/src/test/java/org/elasticsearch/script/python/PythonSecurityTests.java]: Can you expand on this comment to explain how this "gross hack" works? I don't get it at a glance...

**Comment by rjernst** [comment]
LGTM.

File: plugins/lang-python/src/test/java/org/elasticsearch/script/python/PythonSecurityTests.java (hunk 1 above)
Comment (rmuir) [inline plugins/lang-python/src/test/java/org/elasticsearch/script/python/PythonSecurityTests.java]: We try to check that the number system uses ASCII.

**Comment by s1monw** [comment]
it's awesome that we can do that now!!! thanks @rmuir

**Pull request merged**